libraries/WiFi/src/WiFiMulti.cpp

```diff
@@ -120,7 +120,16 @@ uint8_t WiFiMulti::run(uint32_t to) {
             WiFi.beginBSSID(j->ssid, j->bssid);
         }
         while (!WiFi.connected() && (millis() - start < to)) {
-            delay(5);
+            // Replaced delay(5); with the following to prevent
+            // failure when using RTOS
+            unsigned long tWifiDelayForRtos_start = millis();
```

Comment:
`delayMicroseconds(5000); // Avoid task switch in FreeRTOS` does the same thing, more simply.